### PR TITLE
VZ-5808: Customize tests to run against specific release versions

### PR DESCRIPTION
**Pull request by Abhijithang**
- Runs the test stages against specific release versions. 

Fixes VZ-5808

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Comment by kminder** [comment]
We should probably discuss this.  I would have expected that the version gating would need to be more complicated than just equals.  Also it seems it would make more sense for the gate to be in the stage when but I don't know what Jenkins supports there.  Meaning does it support a parameterized function call?

File: ci/upgrade-paths/Jenkinsfile [hunk 1]
```diff
@@ -1050,23 +1051,39 @@ def verifyManagedClusterPermissions() {
...
-def runGinkgoVerifyManaged(testSuitePath, clusterDumpDirectory, skipDeploy, skipUndeploy, skipVerify) {
+def runGinkgoVerifyManaged(testSuitePath, clusterDumpDirectory, skipDeploy, skipUndeploy, skipVerify, String supportedVersion="latest") {
     script {
         int clusterCount = params.TOTAL_CLUSTERS.toInteger()
-        sh """
+        def MANAGED_KUBECONFIG = "${KUBECONFIG_DIR}/2/kube_config"
+        RUNNING_OPERATOR_VERSION = sh(returnStdout: true, script: "export MANAGED_KUBECONFIG='${KUBECONFIG_DIR}/2/kube_config' && kubectl --kubeconfig ${MANAGED_KUBECONFIG} get vz -o jsonpath='{.items[0].status.version}'").trim()
+        echo "RUNNING_OPERATOR_VERSION: ${RUNNING_OPERATOR_VERSION}"
+        def vzVersion = getVZVersion(RUNNING_OPERATOR_VERSION)
+        if(supportedVersion == vzVersion){
```
Comment (pshirodkar-orcl) [inline ci/upgrade-paths/Jenkinsfile:1061]: If...else blocks only need to encompass the ginkgo commands. We do not duplicate unnecessarily other things. Same applies to other similar changes.

File: ci/upgrade-paths/Jenkinsfile [hunk 2]
```diff
@@ -378,24 +379,24 @@ pipeline {
                         }
                         stage ('Deploy Helidon example application - single cluster') {
                             steps {
-                                runGinkgoVerify('examples/helidon', "${TEST_DUMP_ROOT}/helidon-workload", "false", "true", "true", "hello-helidon")
+                                runGinkgoVerify('examples/helidon', "${TEST_DUMP_ROOT}/helidon-workload", "false", "true", "true", "hello-helidon", "v1.2.0")
```
Comment (pshirodkar-orcl) [inline ci/upgrade-paths/Jenkinsfile:382]: Are not going to support versions prior v1.2.0? If not, please update the default value of parameter `VERSION_FOR_INSTALL` (current default value is v1.1.2).

File: ci/upgrade-paths/Jenkinsfile [hunk 3]
```diff
@@ -1050,23 +1051,39 @@ def verifyManagedClusterPermissions() {
...
+def runGinkgoVerifyManaged(testSuitePath, clusterDumpDirectory, skipDeploy, skipUndeploy, skipVerify, String supportedVersion="latest") {
     script {
         int clusterCount = params.TOTAL_CLUSTERS.toInteger()
-        sh """
+        def MANAGED_KUBECONFIG = "${KUBECONFIG_DIR}/2/kube_config"
+        RUNNING_OPERATOR_VERSION = sh(returnStdout: true, script: "export MANAGED_KUBECONFIG='${KUBECONFIG_DIR}/2/kube_config' && kubectl --kubeconfig ${MANAGED_KUBECONFIG} get vz -o jsonpath='{.items[0].status.version}'").trim()
+        echo "RUNNING_OPERATOR_VERSION: ${RUNNING_OPERATOR_VERSION}"
+        def vzVersion = getVZVersion(RUNNING_OPERATOR_VERSION)
+        if(supportedVersion == vzVersion){
+            sh """
```
Comment (pshirodkar-orcl) [inline ci/upgrade-paths/Jenkinsfile:1062]: Both if and else blocks are same.

**Review by kminder — APPROVED** [review]
Approved.

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**